sentry-core/src/client.rs

```diff
@@ -121,11 +115,28 @@ impl Client {
         if event.server_name.is_none() {
             event.server_name = self.options.server_name.clone();
         }
-
         if &event.platform == "other" {
             event.platform = "native".into();
         }
 
+        for integration in self.integrations.read().unwrap().values() {
+            let id = event.event_id;
+            event = match integration.process_event(event) {
+                Some(event) => event,
+                None => {
+                    sentry_debug!("integration dropped event {:?}", id);
+                    return None;
+                }
+            }
+        }
+
+        if let Some(scope) = scope {
```

Comment:
Hmm, I think this was intentionally before the other assignments to ensure that we override potentially invalid information. @HazAT what's the general order of applying integrations, scopes, internal stuff, etc.?

Reply:
Event > Scope > Integrations > Internal

Reply:
Makes sense. It will break the assumption of the code comment above though. Do we want to look at debug meta in scope processors? I want to move those to integrations, so the order would be switched.